compiler/rustc_lexer/src/lib.rs

```diff
@@ -66,6 +66,8 @@ pub enum TokenKind {
     Ident,
     /// "r#ident"
     RawIdent,
+    /// `foo#`, `foo'`, `foo"`. Note the tailer is not included.
+    BadPrefix,
```

Comment:
```suggestion
    /// An unknown prefix like `foo#`, `foo'`, `foo"`. Note that only the prefix (`foo`) is included in the token, not the separator (which is lexed as its own distinct token). In Rust 2021 and later, reserved prefixes are reported as errors; in earlier editions, reserved prefixes generate warnings.
    UnknownPrefix,
```

Reply:
> reserved prefixes generate warnings.

I made that lint allow-by-default, so they're only warnings in case of `cargo fix --edition` or `#![warn(rust_2021_compatibility)]`.